Add post about ideas for digital public works (#105)
x-govuk/x-govuk.github.io: app/posts/digital-public-works-ideas.md

---

> title: Ideas for digital public works

Reviewer: `works` or `goods`? I’m not sure on the definition of one versus the other but have seen digital public goods mentioned more.

Reviewer: Oh, I’ve just read the first line 🤦

description: Some thoughts on digital platforms and services that would be helpful for cross-government teams.
date: 2024-03-01
author:
  name: Frankie Roberto
  url: https://github.com/frankieroberto
---

A recent blog post from dxw suggests that [there needs to be more support for digital public works](https://www.dxw.com/2024/01/there-needs-to-be-more-support-for-digital-public-works/), described as platforms and tools which work across government to make it easier and faster to build services.

> 

Reviewer: Perhaps worth adding a definition of what a digital public work (or digital public good) is for those who aren’t aware. Could utilise the descriptions from the [Digital Public Goods Alliance](https://digitalpublicgoods.net/standard/).

Reviewer 2: Agreed, and also maybe splitting into 2 shorter sentences

Author: Thanks!

> I’m also big supporter of these. This unofficial X-GOVUK community is one way to support cross-government collaboration and tools, but many digital public works would need full time teams to support them.

Reviewer: ```suggestion
I’m also big supporter of these. This unofficial X-GOVUK community is one way to support cross-government collaboration and tools, but many digital public works would need full-time teams to support them.
```

To distinguish from time teams that are full.

> I’m also big supporter of these. This unofficial X-GOVUK community is one way to support cross-government collaboration and tools, but many digital public works would need full time teams to support them.

Reviewer 3: ```suggestion
I’m also a big supporter of these. This unofficial X-GOVUK community is one way to support cross-government collaboration and tools, but many digital public works would need full time teams to support them.
```

> There’s lots to be thankful for. The [platforms developed by the Government Digital Service](https://www.gov.uk/service-toolkit#platforms-and-tools) – Notify, Pay, One Login and Forms – are all super useful.

Reviewer 3: ```suggestion
There’s lots to be thankful for. The [platforms developed by the Government Digital Service](https://www.gov.uk/service-toolkit#platforms-and-tools) – Notify, Pay, Forms and One Login – are all super useful.
```

With One Login being two words, it read like forms was part of it.

> Unfortunately there doesn’t currently seem to be an easy way for government departments working on services to spot other opportunities for common platforms and to turn these into funded cross-government teams and services.

Reviewer 3: I feel it's not so much the spotting, as the highlighting / coordinating. We spot opportunities all the time!

Reviewer 2: Agreed - is it an easy way to suggest them?

Author: I don’t think so? (Which is half the point of this post). I think traditionally only GDS have ever built cross-government digital services, and I don’t know if there are any mechanisms for departments to lobby GDS to create new ones?

Perhaps there could also be ways for departments themselves to band together to create cross-department teams to investigate ideas for services?

However, were this to be possible, I do have a list of ideas for what these services could be.

## The ideas

### 1. Analytics

There are plenty of existing commercial and open source tools for this, so I’m not suggesting rebuilding those, but instead picking one and building a service on top of it. 

The team would do the hard work of navigating the trade-offs between privacy, security and usability to give teams the design patterns and technical implementation for collecting consent and then data.

> They might also build a simple web service to help teams view and understand the most important metrics, with comparisons across services.

Reviewer 3: Getting déjà vu of the performance platform here!

Reviewer: But, soft! what light through yonder window breaks?

Author: 😂

### 2. Address lookup

> The [public sector geospatial agreement](https://www.ordnancesurvey.co.uk/customers/public-sector/public-sector-geospatial-agreement) means that all public sector services should have access to high quality databases of UK addresses. However that doesn’t mean they are easy to integrate with, and some services still end up using Google or other commercial services.

Reviewer 3: Don't think you need to mention google specifically here.

> A central digital team could help by maintaining an API, design patterns and code to make it super easy for teams to let users lookup and select addresses from postcodes.

Reviewer 3: ```suggestion
A central digital team could help by maintaining an API, design patterns and code to make it super easy for teams to let users lookup and select addresses.
```

Reviewer: I agree.

### 3. Hosting

This might have been tried once before, with [PaaS then being decommissioned](https://gds.blog.gov.uk/2022/07/12/why-weve-decided-to-decommission-gov-uk-paas-platform-as-a-service/), but it’s still a good idea.

A rebooted service could be simpler but more opinionated, with the benefit of taking care of more of the low-level issues like operating systems, backups, monitoring and scaling.

> It could also support easy hosting of prototypes.

Reviewer: Wondering if this is the main use case now? If not, it’d be good to add some examples of other use cases above, e.g. which type of teams / orgs / service have those low-level needs? Who would benefit?

Will help make it a little clearer.

### 4. Certifications

There are lots of services where the outcome is that the user receives a certificate or qualification to do something, whether that’s teaching children in schools or working on gas boilers.

Currently, each of these services needs to devise their own way for users to be able to prove or check these certifications.

A central service could allow users to collect all their certifications on a single account and selectively share these with others, either via a code or some other mechanism.

> ### 5. Registers

Reviewer: I’m pretty sure there’s legislation that says this sort of thing couldn’t be built and maintained centrally. It’s also a massive risk to privacy, security and politics (dictators love central databases of people). That’s why models of data exchange for identity platforms look at decentralised storage and re-use, sharing verified credentials or facts rather than having one big database. 

So the technology for this sort of thing is likely to be built into One Login rather than be its own platform.

Author: I've tweaked the paragraph to make it clearer it's about public registers, rather than ones where the register is held by government but not published.

> Unlike the previous version, this could be a platform for the many services which need to maintain public information such as registered landlords, childminders or building inspectors.

Reviewer 3: Link to previous version?

> Unlike the previous version, this could be a platform for the many services which need to maintain public information such as registered landlords, childminders or building inspectors.

Reviewer 3: ```suggestion
Unlike the previous version, this could be a platform for the many existing services which maintain public information such as registered landlords, childminders or building inspectors.
```

There would be tools for organisations to maintain the data, as well as ways for the public to browse and search it. 

> Large-scale registers, such the company register, would still need their own bespoke service, but for small and medium registers a platform would allow these to be built far more quickly and easily.

Reviewer 3: Link to company register

### 6. Monthly payments

> We have GOV.UK Pay for one-off and recurring payments. A possible future extension of this might be users having a single monthly payment to government which includes all the services or licences they currently subscribe to?

Reviewer 3: ```suggestion
We have GOV.UK Pay for one-off and recurring payments. A possible future extension of this might be users having a single monthly payment to government which includes all the services or licences they currently subscribe to.
```

This might make it easier for more services to switch from charging a upfront payment every year or so, allowing users to spread the cost and avoid forgetting to renew.

> ### 7. Statistics publishing

Reviewer: ```suggestion
### 7. Statistics and data publishing
```
Not all data are statistics.

The Office for National Statistics (ONS) has a whole platform for this, but central government also publish a lot of official stats separately on GOV.UK. Currently these are most published as Excel files, PDFs or in other file formats.

Several departments have looked into building platforms for publishing statistics in a more usable and accessible way, and a shared service could be helpful.

It might make sense for this to start by focusing on statistics which are published monthly or quarterly and looking at ways to improve the automation of these.

### 8. Consultations

> Whilst not directly connected to digital services, all government departments and agencies run consultations on proposed policies. The GOV.UK platform lets organisations publish documents for consultation, but they usually use commercial third parties for the consultation itself, where the public can answer questions or upload evidence.

Reviewer: Or, the existing GOV.‌UK platform could be extended to support the features you’re suggesting. It’s already funded, after all!

A central service for running consultations might not only save organisations money, but also help them to run more effective consultations leading to better policy outcomes.

The platform could also make it easier for the public to be discover or be notified about consultations that might affect them, and to find out about the outcomes of consultations they’ve contributed to.

## Closing thoughts

These are just my ideas, based on the work and conversations I’ve been part of.

> I’m interested in what your ideas might be - and whether there could be a way for teams to share these ideas and collaboratively explore whether some of them have the potential to become shared services. 

Reviewer: How do people share their ideas with yo? What’s the call-to-action?

Author: Have re-worked the final para a bit to end with a (slightly rhetorical) question. Hopefully the post will inspire some conversation on LinkedIn/BlueSky/whatever. Not sure what happens from there - we'll see!

I’m also big supporter of these. This unofficial X-GOVUK community is one way to support cross-government collaboration and tools, but many digital public works would need full time teams to support them.

There’s lots to be thankful for. The [platforms developed by the Government Digital Service](https://www.gov.uk/service-toolkit#platforms-and-tools) – Notify, Pay, One Login and Forms – are all super useful.

Unfortunately there doesn’t currently seem to be an easy way for government departments working on services to spot other opportunities for common platforms and to turn these into funded cross-government teams and services.

However, were this to be possible, I do have a list of ideas for what these services could be.

## The ideas

### 1. Analytics

There are plenty of existing commercial and open source tools for this, so I’m not suggesting rebuilding those, but instead picking one and building a service on top of it.

The team would do the hard work of navigating the trade-offs between privacy, security and usability to give teams the design patterns and technical implementation for collecting consent and then data.

> They might also build a simple web service to help teams view and understand the most important metrics, with comparisons across services.

Reviewer 2: I would love this, having just tried to navigate setting up analytics.
